### PR TITLE
Rename data_sequence_number to sequence_number in ManifestEntry

### PR DESCRIPTION
**Pull request by soumya-ghosh**
Implements: https://github.com/apache/iceberg-python/issues/893

### CODE REVIEW

File: pyiceberg/manifest.py [hunk 1]
```diff
@@ -762,10 +762,10 @@ def add_entry(self, entry: ManifestEntry) -> ManifestWriter:
 
         if (
             (entry.status == ManifestEntryStatus.ADDED or entry.status == ManifestEntryStatus.EXISTING)
-            and entry.data_sequence_number is not None
-            and (self._min_data_sequence_number is None or entry.data_sequence_number < self._min_data_sequence_number)
+            and entry.sequence_number is not None
+            and (self._min_data_sequence_number is None or entry.sequence_number < self._min_data_sequence_number)
```
Comment (kevinjqliu) [inline pyiceberg/manifest.py:766]: nit: there are a few variables named `min_data_sequence_number`, should we change those to match? 
https://github.com/search?q=repo%3Aapache%2Ficeberg-python%20data_sequence_number&type=code

File: tests/utils/test_manifest.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def test_read_manifest_entry(generated_manifest_entry_file: str) -> None:
 
     assert manifest_entry.status == ManifestEntryStatus.ADDED
```
Comment (kevinjqliu) [inline tests/utils/test_manifest.py:67]: nit: add a test to check the string representation fo manifest_entry to show that the new field is called `sequence_number`

**Review by kevinjqliu — COMMENTED** [review]
left some nit comments

File: pyiceberg/manifest.py (hunk 1 above)
Comment (soumya-ghosh) [reply pyiceberg/manifest.py:766]: That thought did cross my mind and then I checked the manifest list files and in the actual files it is correct (i.e. `sequence_number` and `min_sequence_number`). 
We only have (private) variable `_min_data_sequence_number` in `ManifestWriter` class, so I left it as it is. 
I guess we should standardize them and make the variable names same as fields in spec.

Comment (kevinjqliu) [reply pyiceberg/manifest.py:766]: i think standardizing them would make it easier to associate the two variables in the future

File: tests/utils/test_manifest.py (hunk 2 above)
Comment (soumya-ghosh) [reply tests/utils/test_manifest.py:67]: I think this is already getting validated in [here](https://github.com/apache/iceberg-python/pull/900/files#diff-9a5a57e7ec723cc4ab6056ce1921f291a501158322d2e6ebf817d568c9621f4bL187).
At the end of the test we are comparing the contents of avro file with the expected `ManifestEntry` object.
Please correct me if my understanding is wrong.

File: pyiceberg/manifest.py (hunk 1 above)
Comment (soumya-ghosh) [reply pyiceberg/manifest.py]: Renamed _min_data_sequence_number to _min_sequence_number in ManifestWriter.

File: tests/utils/test_manifest.py (hunk 2 above)
Comment (kevinjqliu) [reply tests/utils/test_manifest.py:67]: ah i see, thank you. it is implicitly checked in that test

**Review by kevinjqliu — APPROVED** [review]
LGTM

File: pyiceberg/manifest.py (hunk 1 above)
Comment (soumya-ghosh) [reply pyiceberg/manifest.py]: Also renamed `min_data_sequence_number` to `min_sequence_number` in init file pyiceberg.table package, no change required in tests as the variable usage was internal.

**Comment by soumya-ghosh** [comment]
@kevinjqliu missed to rename a variable, hence CI checks failed in mypy stage, it is fixed now.

**Review by kevinjqliu — COMMENTED** [review]
I pulled down the PR and did a search in vscode for `data.*sequence.*number`

Looks like the two relevant parts left are 
`_min_data_file_sequence_number` https://github.com/search?q=repo%3Aapache%2Ficeberg-python+_min_data_file_sequence_number&type=code
and the comment `data sequence number`
https://github.com/search?q=repo%3Aapache%2Ficeberg-python+%22data+sequence+number%22&type=code

Would be great to edit those as well, but not blocking

**Comment by soumya-ghosh** [comment]
@kevinjqliu Changes made as per review comment.

**Review by kevinjqliu — APPROVED** [review]
thank you! LGTM!

**Review by HonahX — COMMENTED** [review]
@soumya-ghosh Thanks for taking this. It looks great! Could you please rebase to the main branch? I think the test error is due to some new code in the main branch need to be updated from "data_sequence_number" to "sequence_number"

**Comment by soumya-ghosh** [comment]
@HonahX Changes done to address failures due to upstream code

**Review by HonahX — APPROVED** [review]
Thanks for updating!

**Comment by soumya-ghosh** [comment]
Thanks @kevinjqliu @HonahX for your review comments.

**Pull request merged**